pkg/sink/postgres.go

```diff
@@ -561,29 +583,94 @@ func (p *PGXSink) handleUpdate(ctx context.Context, m *pb.Change) (err error) {
 		}
 	}
 
-	return p.raw.ExecParams(ctx, sql.UpdateQuery(m.Schema, m.Table, sets, keys), vals, oids, fmts, fmts).Read().Err
+	p.pendingChanges = append(p.pendingChanges, pendingChange{
+		sql:           sql.UpdateQuery(m.Schema, m.Table, sets, keys),
+		args:          vals,
+		paramOIDs:     oids,
+		paramFormats:  fmts,
+		resultFormats: fmts,
+	})
+	return
 }
 
 const (
 	UpdateSourceSQL = "update pgcapture.sources set commit=$1,seq=$2,mid=$3,commit_ts=$4,apply_ts=now() where id=$5"
 )
 
-func (p *PGXSink) handleCommit(cp cursor.Checkpoint, commit *pb.Commit) (err error) {
-	ctx := context.Background()
-	if err = p.flushInsert(ctx); err != nil {
+func (p *PGXSink) handleCommit(sourceRemaining int, cp cursor.Checkpoint, commit *pb.Commit) (err error) {
+	if err = p.flushInsert(); err != nil {
 		return err
 	}
-	if _, err = p.conn.Prepare(ctx, UpdateSourceSQL, UpdateSourceSQL); err != nil {
-		return err
+
+	for _, q := range p.pendingChanges {
+		p.pipeline.SendQueryParams(q.sql, q.args, q.paramOIDs, q.paramFormats, q.resultFormats)
+	}
+	p.pendingChanges = p.pendingChanges[:0]
+	p.pendingCommit = &pendingCommit{
+		checkPoint: cp,
+		commit:     commit,
 	}
-	commitTs := pgTz(commit.CommitTime)
-	if _, err = p.conn.Exec(ctx, UpdateSourceSQL, pgLSN(cp.LSN), pgInt4(int32(cp.Seq)), cp.Data, commitTs, p.pgSrcID); err != nil {
+	p.pendingCommitted = append(p.pendingCommitted, cp)
```

Comment:
We should merge the `pendingCommitted` and the `pendingCommit` into one field, for example
```go
pendingCommits []pendingCommit
```

Otherwise, it is a little hard to find out the purpose of `pendingCommit` and the naming `pendingCommitted` is a little confusing.

Reply:
good idea!

Reply:
[feat: merge pendingCommitted and pendingCommit](https://github.com/rueian/pgcapture/pull/48/commits/4f78fdb1af1ba153a3bbaf34af2cdab0a78628d9)